crates/ty_python_semantic/src/types.rs

```diff
@@ -6227,6 +6247,35 @@ impl<'db> InvalidTypeExpression<'db> {
     }
 }
 
+/// Data regarding a `warnings.deprecated` or `typing_extensions.deprecated` decorator.
+#[salsa::interned(debug)]
+#[derive(PartialOrd, Ord)]
+pub struct DeprecatedInstance<'db> {
+    /// The message for the deprecation
+    pub message: StringLiteralType<'db>,
+
+    /// The `@warnings.deprecated(...)` invocation
+    pub definition: Option<Definition<'db>>,
+}
+
+// The Salsa heap is tracked separately.
+impl get_size2::GetSize for DeprecatedInstance<'_> {}
+
+impl<'db> DeprecatedInstance<'db> {
+    pub(crate) fn normalized_impl(
+        self,
+        db: &'db dyn Db,
+        _visitor: &mut TypeTransformer<'db>,
+    ) -> Self {
+        Self::new(
+            db,
+            // StringLiteralType has no normalization to do
+            self.message(db),
+            self.definition(db),
+        )
```

Comment:
Can this just return `self`?
```suggestion
        self
```

Reply:
The whole function can be removed, I figured it might be good to write it out in-case something gets added but... eh.